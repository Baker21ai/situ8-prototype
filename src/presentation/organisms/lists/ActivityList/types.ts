/**
 * TypeScript interfaces for ActivityList compound component
 */

import { EnterpriseActivity, ActivityCluster } from '../../../../../lib/types/activity';
import { Priority } from '../../../../../lib/utils/status';
import { EnterpriseCardVariant } from '../../../../../components/EnterpriseActivityCard';

// Filter state for activities
export interface ActivityFilterState {
  search: string;
  priorities: Priority[];
  statuses: string[];
  types: string[];
  locations: string[];
  timeRange: 'live' | '15m' | '1h' | '4h' | '24h' | 'custom';
  businessImpact: string[];
  confidenceThreshold: number;
  showClusters: boolean;
  aiFiltering: boolean;
}

// Performance metrics
export interface ActivityPerformanceMetrics {
  totalActivities: number;
  renderTime: number;
  filteredCount: number;
  criticalCount: number;
  averageConfidence: number;
  processingRate: number;
}

// Layout modes
export type ActivityLayoutMode = 'grid' | 'horizontal';
export type ActivityViewMode = 'stream' | 'minimal' | 'summary';

// Context data for compound component
export interface ActivityListContextData {
  // Data
  activities: (EnterpriseActivity | ActivityCluster)[];
  filteredActivities: (EnterpriseActivity | ActivityCluster)[];
  
  // State
  filters: ActivityFilterState;
  selectedItems: Set<string>;
  viewMode: ActivityViewMode;
  layoutMode: ActivityLayoutMode;
  compactMode: boolean;
  useVirtualScrolling: boolean;
  showPerformanceMetrics: boolean;
  
  // Metrics
  performanceMetrics: ActivityPerformanceMetrics;
  priorityMetrics: Record<Priority, number>;
  
  // Event handlers
  onActivitySelect?: (activity: EnterpriseActivity | ActivityCluster) => void;
  onActivityAction?: (action: string, activity: EnterpriseActivity | ActivityCluster) => void;
  onBulkAction?: (action: string, activities: EnterpriseActivity[]) => void;
  
  // State setters
  setFilters: (filters: Partial<ActivityFilterState>) => void;
  setSelectedItems: (items: Set<string>) => void;
  setViewMode: (mode: ActivityViewMode) => void;
  setLayoutMode: (mode: ActivityLayoutMode) => void;
  setCompactMode: (compact: boolean) => void;
  setUseVirtualScrolling: (enabled: boolean) => void;
  setShowPerformanceMetrics: (show: boolean) => void;
  
  // Derived data
  hasSelection: boolean;
  isEmpty: boolean;
  
  // Configuration
  realTimeMode?: boolean;
  height?: number;
  className?: string;
}

// Component props
export interface ActivityListProps {
  activities: (EnterpriseActivity | ActivityCluster)[];
  onActivitySelect?: (activity: EnterpriseActivity | ActivityCluster) => void;
  onActivityAction?: (action: string, activity: EnterpriseActivity | ActivityCluster) => void;
  onBulkAction?: (action: string, activities: EnterpriseActivity[]) => void;
  realTimeMode?: boolean;
  className?: string;
  height?: number;
  children?: React.ReactNode;
}

// Header component props
export interface ActivityListHeaderProps {
  children?: React.ReactNode;
  className?: string;
}

// Content component props
export interface ActivityListContentProps {
  className?: string;
  height?: number;
}

// Footer component props
export interface ActivityListFooterProps {
  className?: string;
}

// Filters component props
export interface ActivityListFiltersProps {
  className?: string;
  showAdvanced?: boolean;
}

// Stats component props
export interface ActivityListStatsProps {
  className?: string;
  detailed?: boolean;
}

// Search component props
export interface ActivityListSearchProps {
  placeholder?: string;
  className?: string;
}

// View toggle component props
export interface ActivityListViewToggleProps {
  className?: string;
}

// Priority segment for display
export interface PrioritySegment {
  priority: Priority;
  items: (EnterpriseActivity | ActivityCluster)[];
  count: number;
  color: {
    bg: string;
    border: string;
    text: string;
  };
  icon: React.ReactNode;
}

// Compound component type with sub-components
export interface ActivityListCompound {
  (props: ActivityListProps): JSX.Element;
  Header: React.ComponentType<ActivityListHeaderProps>;
  Content: React.ComponentType<ActivityListContentProps>;
  Footer: React.ComponentType<ActivityListFooterProps>;
  Filters: React.ComponentType<ActivityListFiltersProps>;
  Stats: React.ComponentType<ActivityListStatsProps>;
  Search: React.ComponentType<ActivityListSearchProps>;
  ViewToggle: React.ComponentType<ActivityListViewToggleProps>;
}