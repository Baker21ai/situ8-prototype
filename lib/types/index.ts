/**
 * Central export for all type definitions
 */

// Re-export all activity types
export * from './activity';

// Re-export all guard types
export * from './guards';

// Re-export all communication types
export * from './communications';

// Re-export utility types
export type { Priority, Status, BusinessImpact } from '../utils/status';
export type { ActivityType, ThreatLevel, SecurityLevel } from '../utils/security';

// Common UI component prop types
export interface BaseComponentProps {
  className?: string;
  children?: React.ReactNode;
}

export interface InteractiveComponentProps extends BaseComponentProps {
  onClick?: () => void;
  onHover?: () => void;
  disabled?: boolean;
}

// Component size variants
export type ComponentSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';

// Component variants
export type ComponentVariant = 'default' | 'outline' | 'ghost' | 'destructive' | 'success';

// Layout variants
export type LayoutVariant = 'compact' | 'comfortable' | 'spacious';

// View modes
export type ViewMode = 'grid' | 'list' | 'table' | 'kanban' | 'timeline';

// Sort options
export interface SortOption<T = string> {
  field: T;
  direction: 'asc' | 'desc';
  label: string;
}

// Pagination
export interface PaginationState {
  page: number;
  pageSize: number;
  total: number;
  totalPages: number;
}

// Selection state
export interface SelectionState<T = string> {
  selected: Set<T>;
  lastSelected?: T;
  mode: 'single' | 'multiple';
}

// Filter state
export interface FilterState {
  active: boolean;
  filters: Record<string, any>;
  quickFilters?: string[];
}

// Search state
export interface SearchState {
  query: string;
  isSearching: boolean;
  results?: any[];
  resultCount?: number;
}

// Modal/Dialog props
export interface ModalProps extends BaseComponentProps {
  isOpen: boolean;
  onClose: () => void;
  title?: string;
  description?: string;
  size?: 'sm' | 'md' | 'lg' | 'xl' | 'full';
}

// Form field props
export interface FormFieldProps<T = string> {
  name: string;
  label?: string;
  value: T;
  onChange: (value: T) => void;
  error?: string;
  required?: boolean;
  disabled?: boolean;
  placeholder?: string;
}

// Action handler types
export type ActionHandler<T = void> = (data: T) => void | Promise<void>;
export type SelectHandler<T> = (item: T) => void;
export type MultiSelectHandler<T> = (items: T[]) => void;

// API response types
export interface ApiResponse<T> {
  data: T;
  success: boolean;
  message?: string;
  error?: string;
  timestamp: Date;
}

export interface PaginatedResponse<T> extends ApiResponse<T[]> {
  pagination: PaginationState;
}

// Error types
export interface AppError {
  code: string;
  message: string;
  details?: Record<string, any>;
  timestamp: Date;
}

// Notification types
export interface Notification {
  id: string;
  type: 'info' | 'success' | 'warning' | 'error';
  title: string;
  message?: string;
  timestamp: Date;
  duration?: number;
  action?: {
    label: string;
    handler: () => void;
  };
}