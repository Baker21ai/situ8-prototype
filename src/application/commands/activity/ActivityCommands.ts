/**
 * Activity Domain Commands
 * All commands related to activity management
 */

import { ICommand } from '../base/ICommand';
import { ActivityType } from '../../../lib/utils/security';
import { Priority, Status } from '../../../lib/utils/status';

// ===== CREATE ACTIVITY COMMAND =====

export interface CreateActivityCommand extends ICommand {
  readonly type: 'CreateActivity';
  readonly data: {
    activityType: ActivityType;
    title: string;
    location: string;
    priority: Priority;
    description?: string;
    building?: string;
    zone?: string;
    assignedTo?: string;
    confidence?: number;
    externalData?: {
      sourceSystem: string;
      originalType: string;
      rawPayload: Record<string, any>;
    };
  };
}

export interface CreateActivityResult {
  activityId: string;
  autoTagsApplied: string[];
  incidentTriggered?: {
    incidentId: string;
    rule: string;
  };
}

// ===== UPDATE ACTIVITY COMMAND =====

export interface UpdateActivityCommand extends ICommand {
  readonly type: 'UpdateActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    updates: {
      priority?: Priority;
      status?: Status;
      description?: string;
      assignedTo?: string;
      userTags?: string[];
    };
    reason?: string;
  };
}

export interface UpdateActivityResult {
  updatedFields: string[];
  triggeredWorkflows?: string[];
}

// ===== ASSIGN ACTIVITY COMMAND =====

export interface AssignActivityCommand extends ICommand {
  readonly type: 'AssignActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    assignedTo: string;
    assignedBy: string;
    reason?: string;
    notifyAssignee?: boolean;
  };
}

export interface AssignActivityResult {
  previousAssignee?: string;
  notificationSent: boolean;
}

// ===== ARCHIVE ACTIVITY COMMAND =====

export interface ArchiveActivityCommand extends ICommand {
  readonly type: 'ArchiveActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    reason: string;
    archivedBy: string;
    permanent?: boolean;
  };
}

export interface ArchiveActivityResult {
  archivedAt: Date;
  retentionDate: Date;
}

// ===== BULK STATUS UPDATE COMMAND =====

export interface BulkUpdateStatusCommand extends ICommand {
  readonly type: 'BulkUpdateStatus';
  readonly data: {
    activityIds: string[];
    status: Status;
    reason?: string;
    notifyAssigned?: boolean;
  };
}

export interface BulkUpdateStatusResult {
  successCount: number;
  failedIds: string[];
  notificationsSent: number;
}

// ===== ADD USER TAG COMMAND =====

export interface AddUserTagCommand extends ICommand {
  readonly type: 'AddUserTag';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    tag: string;
    addedBy: string;
  };
}

export interface AddUserTagResult {
  tag: string;
  allTags: string[];
}

// ===== REMOVE USER TAG COMMAND =====

export interface RemoveUserTagCommand extends ICommand {
  readonly type: 'RemoveUserTag';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    tag: string;
    removedBy: string;
  };
}

export interface RemoveUserTagResult {
  tag: string;
  allTags: string[];
}

// ===== ESCALATE ACTIVITY COMMAND =====

export interface EscalateActivityCommand extends ICommand {
  readonly type: 'EscalateActivity';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    escalationType: 'priority' | 'assignment' | 'incident';
    targetPriority?: Priority;
    targetAssignee?: string;
    createIncident?: boolean;
    reason: string;
    escalatedBy: string;
  };
}

export interface EscalateActivityResult {
  escalationType: string;
  previousState: {
    priority?: Priority;
    assignedTo?: string;
  };
  newState: {
    priority?: Priority;
    assignedTo?: string;
    incidentId?: string;
  };
}

// ===== LINK TO INCIDENT COMMAND =====

export interface LinkToIncidentCommand extends ICommand {
  readonly type: 'LinkToIncident';
  readonly aggregateId: string; // Activity ID
  readonly data: {
    incidentId: string;
    linkType: 'trigger' | 'related' | 'evidence';
    linkedBy: string;
    notes?: string;
  };
}

export interface LinkToIncidentResult {
  incidentId: string;
  linkType: string;
  allIncidentLinks: string[];
}

// ===== BATCH CREATE ACTIVITIES COMMAND =====

export interface BatchCreateActivitiesCommand extends ICommand {
  readonly type: 'BatchCreateActivities';
  readonly data: {
    activities: Array<{
      activityType: ActivityType;
      title: string;
      location: string;
      priority: Priority;
      description?: string;
      building?: string;
      zone?: string;
      confidence?: number;
      externalData?: {
        sourceSystem: string;
        originalType: string;
        rawPayload: Record<string, any>;
      };
    }>;
    batchId: string;
    sourceSystem?: string;
  };
}

export interface BatchCreateActivitiesResult {
  batchId: string;
  successCount: number;
  failedCount: number;
  createdActivityIds: string[];
  failures: Array<{
    index: number;
    error: string;
  }>;
  incidentsTriggered: Array<{
    activityId: string;
    incidentId: string;
    rule: string;
  }>;
}

// ===== MERGE ACTIVITIES COMMAND =====

export interface MergeActivitiesCommand extends ICommand {
  readonly type: 'MergeActivities';
  readonly data: {
    primaryActivityId: string;
    activityIdsToMerge: string[];
    mergeReason: string;
    mergedBy: string;
    preserveData?: {
      tags: boolean;
      assignments: boolean;
      timeline: boolean;
    };
  };
}

export interface MergeActivitiesResult {
  primaryActivityId: string;
  mergedActivityIds: string[];
  mergedData: {
    tagsAdded: string[];
    timelineEntriesAdded: number;
    evidenceLinked: string[];
  };
}

// Union type of all activity commands
export type ActivityCommand = 
  | CreateActivityCommand
  | UpdateActivityCommand
  | AssignActivityCommand
  | ArchiveActivityCommand
  | BulkUpdateStatusCommand
  | AddUserTagCommand
  | RemoveUserTagCommand
  | EscalateActivityCommand
  | LinkToIncidentCommand
  | BatchCreateActivitiesCommand
  | MergeActivitiesCommand;