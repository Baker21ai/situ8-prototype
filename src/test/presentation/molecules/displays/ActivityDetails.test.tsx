import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { 
  ActivityDetails, 
  ActivityDetailsSummary,
  ActivityDetailsCompact,
  ActivityDetailsInline,
  ActivityDetailsProps 
} from '../../../../presentation/molecules/displays/ActivityDetails';
import { EnterpriseActivity, ActivityCluster } from '../../../../../lib/types/activity';

// Mock atomic components
vi.mock('../../../../presentation/atoms/badges/StatusBadge', () => ({
  StatusBadge: ({ priority, status, size }: any) => (
    <div data-testid="status-badge">{status} ({priority}, {size})</div>
  )
}));

vi.mock('../../../../presentation/atoms/indicators/PriorityIndicator', () => ({
  PriorityIndicator: ({ priority, size, variant }: any) => (
    <div data-testid="priority-indicator">{priority} ({size}, {variant})</div>
  )
}));

vi.mock('../../../../presentation/atoms/badges/LocationTag', () => ({
  LocationTag: ({ location, type, size, showIcon }: any) => (
    <div data-testid="location-tag">{location} ({type}, {size}, icon: {showIcon.toString()})</div>
  )
}));

vi.mock('../../../../presentation/atoms/displays/TimeDisplay', () => ({
  TimeDisplay: ({ date, format, size, showIcon }: any) => (
    <div data-testid="time-display">{format} ({size}, icon: {showIcon?.toString() || 'false'})</div>
  )
}));

// Mock UI components
vi.mock('../../../../../components/ui/badge', () => ({
  Badge: ({ children, variant, size, className }: any) => (
    <div data-testid="badge" className={className}>{children}</div>
  )
}));

// Mock Lucide icons
vi.mock('lucide-react', () => ({
  AlertTriangle: () => <div data-testid="alert-triangle-icon" />,
  Users: () => <div data-testid="users-icon" />,
  Camera: () => <div data-testid="camera-icon" />,
  Tag: () => <div data-testid="tag-icon" />
}));

describe('ActivityDetails', () => {
  const mockActivity: EnterpriseActivity = {
    id: '1',
    title: 'Security Alert',
    description: 'Unauthorized access detected',
    type: 'security-breach',
    priority: 'high',
    status: 'detecting',
    timestamp: '2024-01-15T14:30:00Z',
    location: 'Building A',
    zone: 'Zone 1',
    assignedTo: 'John Doe',
    respondingUnits: ['Unit 1', 'Unit 2'],
    cameraName: 'Camera 1',
    additionalCameras: ['Camera 2', 'Camera 3'],
    tags: ['security', 'urgent'],
    confidence: 0.85
  };

  const mockCluster: ActivityCluster = {
    id: 'cluster-1',
    clusterType: 'cluster',
    title: '5 activities in Building A',
    description: 'Multiple security events detected',
    location: 'Building A',
    timestamp: '2024-01-15T14:30:00Z',
    lastActivity: '2024-01-15T15:00:00Z',
    count: 5,
    highestPriority: 'critical',
    types: ['security-breach', 'alert'],
    timeRange: '30 minutes',
    activities: [mockActivity]
  };

  const defaultProps: ActivityDetailsProps = {
    activity: mockActivity
  };

  describe('Rendering', () => {
    it('should render with default props', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      expect(screen.getByText('Security Alert')).toBeInTheDocument();
      expect(screen.getByText('Unauthorized access detected')).toBeInTheDocument();
    });

    it('should render cluster information', () => {
      render(<ActivityDetails activity={mockCluster} />);
      
      expect(screen.getByText('5 activities in Building A')).toBeInTheDocument();
      expect(screen.getByText('5 activities')).toBeInTheDocument();
    });
  });

  describe('Variant styles', () => {
    it('should apply full variant styles (default)', () => {
      const { container } = render(<ActivityDetails {...defaultProps} variant="full" />);
      
      const mainContainer = container.firstChild as HTMLElement;
      expect(mainContainer.className).toContain('space-y-3');
    });

    it('should apply summary variant styles', () => {
      const { container } = render(<ActivityDetails {...defaultProps} variant="summary" />);
      
      const mainContainer = container.firstChild as HTMLElement;
      expect(mainContainer.className).toContain('space-y-2');
    });

    it('should apply compact variant styles', () => {
      const { container } = render(<ActivityDetails {...defaultProps} variant="compact" />);
      
      const mainContainer = container.firstChild as HTMLElement;
      expect(mainContainer.className).toContain('space-y-1');
    });

    it('should apply inline variant styles', () => {
      const { container } = render(<ActivityDetails {...defaultProps} variant="inline" />);
      
      const mainContainer = container.firstChild as HTMLElement;
      expect(mainContainer.className).toContain('space-x-2');
      expect(mainContainer.className).toContain('flex');
      expect(mainContainer.className).toContain('items-center');
    });
  });

  describe('Content display options', () => {
    it('should show metadata when showMetadata is true (default)', () => {
      render(<ActivityDetails {...defaultProps} showMetadata={true} />);
      
      expect(screen.getByTestId('priority-indicator')).toBeInTheDocument();
      expect(screen.getByTestId('status-badge')).toBeInTheDocument();
      expect(screen.getByTestId('location-tag')).toBeInTheDocument();
      expect(screen.getByTestId('time-display')).toBeInTheDocument();
    });

    it('should hide metadata when showMetadata is false', () => {
      render(<ActivityDetails {...defaultProps} showMetadata={false} />);
      
      expect(screen.queryByTestId('priority-indicator')).not.toBeInTheDocument();
      expect(screen.queryByTestId('status-badge')).not.toBeInTheDocument();
      expect(screen.queryByTestId('location-tag')).not.toBeInTheDocument();
      expect(screen.queryByTestId('time-display')).not.toBeInTheDocument();
    });

    it('should show description when showDescription is true (default)', () => {
      render(<ActivityDetails {...defaultProps} showDescription={true} />);
      
      expect(screen.getByText('Unauthorized access detected')).toBeInTheDocument();
    });

    it('should hide description when showDescription is false', () => {
      render(<ActivityDetails {...defaultProps} showDescription={false} />);
      
      expect(screen.queryByText('Unauthorized access detected')).not.toBeInTheDocument();
    });

    it('should show assignment when showAssignment is true (default)', () => {
      render(<ActivityDetails {...defaultProps} showAssignment={true} />);
      
      expect(screen.getByText(/Assigned to: John Doe/)).toBeInTheDocument();
      expect(screen.getByText(/2 units responding/)).toBeInTheDocument();
    });

    it('should hide assignment when showAssignment is false', () => {
      render(<ActivityDetails {...defaultProps} showAssignment={false} />);
      
      expect(screen.queryByText(/Assigned to:/)).not.toBeInTheDocument();
    });

    it('should show cameras when showCameras is true (default)', () => {
      render(<ActivityDetails {...defaultProps} showCameras={true} />);
      
      expect(screen.getByText('Camera 1')).toBeInTheDocument();
      expect(screen.getByText('+2 more')).toBeInTheDocument();
    });

    it('should hide cameras when showCameras is false', () => {
      render(<ActivityDetails {...defaultProps} showCameras={false} />);
      
      expect(screen.queryByText('Camera 1')).not.toBeInTheDocument();
    });

    it('should show tags when showTags is true (default)', () => {
      render(<ActivityDetails {...defaultProps} showTags={true} />);
      
      expect(screen.getByText('security')).toBeInTheDocument();
      expect(screen.getByText('urgent')).toBeInTheDocument();
    });

    it('should hide tags when showTags is false', () => {
      render(<ActivityDetails {...defaultProps} showTags={false} />);
      
      expect(screen.queryByText('security')).not.toBeInTheDocument();
      expect(screen.queryByText('urgent')).not.toBeInTheDocument();
    });
  });

  describe('Cluster-specific rendering', () => {
    it('should render cluster count badge', () => {
      render(<ActivityDetails activity={mockCluster} />);
      
      const badges = screen.getAllByTestId('badge');
      const countBadge = badges.find(badge => badge.textContent?.includes('5 activities'));
      expect(countBadge).toBeInTheDocument();
    });

    it('should not show assignment info for clusters', () => {
      render(<ActivityDetails activity={mockCluster} />);
      
      expect(screen.queryByText(/Assigned to:/)).not.toBeInTheDocument();
    });

    it('should not show camera info for clusters', () => {
      render(<ActivityDetails activity={mockCluster} />);
      
      expect(screen.queryByText('Camera 1')).not.toBeInTheDocument();
    });

    it('should not show tags for clusters', () => {
      render(<ActivityDetails activity={mockCluster} />);
      
      expect(screen.queryByText('security')).not.toBeInTheDocument();
    });

    it('should show cluster-specific metadata', () => {
      render(<ActivityDetails activity={mockCluster} />);
      
      expect(screen.getByText(/Time Range: 30 minutes/)).toBeInTheDocument();
      expect(screen.getByText(/Highest Priority: critical/)).toBeInTheDocument();
      expect(screen.getByText(/Types: security-breach, alert/)).toBeInTheDocument();
    });
  });

  describe('Special activity types', () => {
    it('should show confidence for alert activities', () => {
      const alertActivity = { ...mockActivity, type: 'alert', confidence: 0.85 };
      render(<ActivityDetails activity={alertActivity} />);
      
      expect(screen.getByText('Confidence: 85%')).toBeInTheDocument();
    });

    it('should not show confidence for non-alert activities', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      expect(screen.queryByText(/Confidence:/)).not.toBeInTheDocument();
    });
  });

  describe('Component integration', () => {
    it('should pass correct props to StatusBadge', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      const statusBadge = screen.getByTestId('status-badge');
      expect(statusBadge).toHaveTextContent('detecting (high, md)');
    });

    it('should pass correct props to PriorityIndicator', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      const priorityIndicator = screen.getByTestId('priority-indicator');
      expect(priorityIndicator).toHaveTextContent('high (md, badge)');
    });

    it('should pass correct props to LocationTag', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      const locationTag = screen.getByTestId('location-tag');
      expect(locationTag).toHaveTextContent('Building A (site, md, icon: true)');
    });

    it('should pass correct props to TimeDisplay', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      const timeDisplay = screen.getByTestId('time-display');
      expect(timeDisplay).toHaveTextContent('relative (md, icon: true)');
    });
  });

  describe('Size adaptations', () => {
    it('should use correct badge sizes for compact variant', () => {
      render(<ActivityDetails {...defaultProps} variant="compact" />);
      
      const statusBadge = screen.getByTestId('status-badge');
      expect(statusBadge).toHaveTextContent('detecting (high, xs)');
    });

    it('should use correct badge sizes for inline variant', () => {
      render(<ActivityDetails {...defaultProps} variant="inline" />);
      
      const statusBadge = screen.getByTestId('status-badge');
      expect(statusBadge).toHaveTextContent('detecting (high, xs)');
    });

    it('should use correct badge sizes for summary variant', () => {
      render(<ActivityDetails {...defaultProps} variant="summary" />);
      
      const statusBadge = screen.getByTestId('status-badge');
      expect(statusBadge).toHaveTextContent('detecting (high, sm)');
    });
  });

  describe('Custom styling', () => {
    it('should apply custom className', () => {
      const customClass = 'custom-activity-details';
      render(<ActivityDetails {...defaultProps} className={customClass} />);
      
      const container = screen.getByText('Security Alert').closest('div');
      expect(container?.className).toContain(customClass);
    });
  });

  describe('Edge cases', () => {
    it('should handle activity without description', () => {
      const activityWithoutDescription = { ...mockActivity, description: undefined };
      render(<ActivityDetails activity={activityWithoutDescription} />);
      
      expect(screen.getByText('Security Alert')).toBeInTheDocument();
      expect(screen.queryByText('Unauthorized access detected')).not.toBeInTheDocument();
    });

    it('should handle activity without assignment', () => {
      const unassignedActivity = { ...mockActivity, assignedTo: undefined, respondingUnits: undefined };
      render(<ActivityDetails activity={unassignedActivity} />);
      
      expect(screen.queryByText(/Assigned to:/)).not.toBeInTheDocument();
    });

    it('should handle activity without cameras', () => {
      const activityWithoutCameras = { ...mockActivity, cameraName: undefined, additionalCameras: undefined };
      render(<ActivityDetails activity={activityWithoutCameras} />);
      
      expect(screen.queryByText('Camera 1')).not.toBeInTheDocument();
    });

    it('should handle activity without tags', () => {
      const activityWithoutTags = { ...mockActivity, tags: undefined };
      render(<ActivityDetails activity={activityWithoutTags} />);
      
      expect(screen.queryByText('security')).not.toBeInTheDocument();
    });

    it('should handle empty tags array', () => {
      const activityWithEmptyTags = { ...mockActivity, tags: [] };
      render(<ActivityDetails activity={activityWithEmptyTags} />);
      
      expect(screen.queryByTestId('tag-icon')).not.toBeInTheDocument();
    });
  });

  describe('Specialized variants', () => {
    it('should render ActivityDetailsSummary correctly', () => {
      const { container } = render(<ActivityDetailsSummary activity={mockActivity} />);
      
      const mainContainer = container.firstChild as HTMLElement;
      expect(mainContainer.className).toContain('space-y-2');
    });

    it('should render ActivityDetailsCompact correctly', () => {
      const { container } = render(<ActivityDetailsCompact activity={mockActivity} />);
      
      const mainContainer = container.firstChild as HTMLElement;
      expect(mainContainer.className).toContain('space-y-1');
    });

    it('should render ActivityDetailsInline correctly', () => {
      const { container } = render(<ActivityDetailsInline activity={mockActivity} />);
      
      const mainContainer = container.firstChild as HTMLElement;
      expect(mainContainer.className).toContain('flex');
      expect(mainContainer.className).toContain('items-center');
    });
  });

  describe('Accessibility', () => {
    it('should have proper heading structure', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      const heading = screen.getByRole('heading', { level: 3 });
      expect(heading).toHaveTextContent('Security Alert');
    });

    it('should be keyboard navigable', () => {
      render(<ActivityDetails {...defaultProps} />);
      
      const container = screen.getByText('Security Alert').closest('div');
      expect(container).toBeInTheDocument();
      // Additional keyboard navigation tests would be added here
    });
  });

  describe('Snapshot tests', () => {
    it('should match snapshot for default variant', () => {
      const { container } = render(<ActivityDetails activity={mockActivity} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for cluster', () => {
      const { container } = render(<ActivityDetails activity={mockCluster} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for inline variant', () => {
      const { container } = render(<ActivityDetails activity={mockActivity} variant="inline" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
