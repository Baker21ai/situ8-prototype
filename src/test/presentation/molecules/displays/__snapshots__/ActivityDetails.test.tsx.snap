// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActivityDetails > Snapshot tests > should match snapshot for cluster 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex items-start justify-between"
  >
    <div
      class="flex-1"
    >
      <h3
        class="text-lg font-semibold "
      >
        5 activities in Building A
      </h3>
      <div
        class="ml-2"
        data-testid="badge"
      >
        <div
          data-testid="users-icon"
        />
        5
         activities
      </div>
    </div>
    <div
      class="flex items-center gap-2 flex-shrink-0"
    >
      <div
        data-testid="priority-indicator"
      >
        critical
         (
        md
        , 
        badge
        )
      </div>
    </div>
  </div>
  <div
    class="flex items-center gap-3 text-sm text-muted-foreground"
  >
    <div
      data-testid="location-tag"
    >
      Building A
       (
      site
      , 
      md
      , icon: 
      true
      )
    </div>
    <div
      data-testid="time-display"
    >
      relative
       (
      md
      , icon: 
      true
      )
    </div>
  </div>
  <div
    class="text-sm text-muted-foreground"
  >
    Multiple security events detected
  </div>
  <div
    class="grid grid-cols-2 gap-2 text-xs text-muted-foreground"
  >
    <div>
      Time Range: 
      30 minutes
    </div>
    <div>
      Highest Priority: 
      critical
    </div>
    <div>
      Types: 
      security-breach, alert
    </div>
    <div>
      Last Activity:
      <div
        data-testid="time-display"
      >
        relative
         (
        xs
        , icon: 
        false
        )
      </div>
    </div>
  </div>
</div>
`;

exports[`ActivityDetails > Snapshot tests > should match snapshot for default variant 1`] = `
<div
  class="space-y-3 "
>
  <div
    class="flex items-start justify-between"
  >
    <div
      class="flex-1"
    >
      <h3
        class="text-lg font-semibold "
      >
        Security Alert
      </h3>
    </div>
    <div
      class="flex items-center gap-2 flex-shrink-0"
    >
      <div
        data-testid="priority-indicator"
      >
        high
         (
        md
        , 
        badge
        )
      </div>
      <div
        data-testid="status-badge"
      >
        detecting
         (
        high
        , 
        md
        )
      </div>
    </div>
  </div>
  <div
    class="flex items-center gap-3 text-sm text-muted-foreground"
  >
    <div
      data-testid="location-tag"
    >
      Building A
       (
      site
      , 
      md
      , icon: 
      true
      )
    </div>
    <div
      data-testid="time-display"
    >
      relative
       (
      md
      , icon: 
      true
      )
    </div>
  </div>
  <div
    class="text-sm text-muted-foreground"
  >
    Unauthorized access detected
  </div>
  <div
    class="text-sm text-muted-foreground"
  >
    <div
      class="flex items-center gap-1"
    >
      <div
        data-testid="users-icon"
      />
      <span>
        Assigned to: 
        John Doe
      </span>
      <span
        class="ml-2"
      >
        • 
        2
         units responding
      </span>
    </div>
  </div>
  <div
    class="text-sm text-muted-foreground"
  >
    <div
      class="flex items-center gap-1"
    >
      <div
        data-testid="camera-icon"
      />
      <span>
        Camera 1
      </span>
      <span
        class="ml-2"
      >
        +
        2
         more
      </span>
    </div>
  </div>
  <div
    class="flex items-center gap-1 flex-wrap"
  >
    <div
      data-testid="tag-icon"
    />
    <div
      class="text-xs"
      data-testid="badge"
    >
      security
    </div>
    <div
      class="text-xs"
      data-testid="badge"
    >
      urgent
    </div>
  </div>
</div>
`;

exports[`ActivityDetails > Snapshot tests > should match snapshot for inline variant 1`] = `
<div
  class="space-x-2 flex items-center flex-wrap"
>
  <div
    class="contents"
  >
    <div
      class="contents"
    >
      <h3
        class="text-sm font-medium mr-2"
      >
        Security Alert
      </h3>
    </div>
    <div
      class="flex items-center gap-2 "
    >
      <div
        data-testid="priority-indicator"
      >
        high
         (
        xs
        , 
        badge
        )
      </div>
      <div
        data-testid="status-badge"
      >
        detecting
         (
        high
        , 
        xs
        )
      </div>
    </div>
  </div>
  <div
    class="flex items-center gap-3 text-xs text-muted-foreground"
  >
    <div
      data-testid="location-tag"
    >
      Building A
       (
      site
      , 
      xs
      , icon: 
      true
      )
    </div>
    <div
      data-testid="time-display"
    >
      relative
       (
      xs
      , icon: 
      true
      )
    </div>
  </div>
  <div
    class="text-xs text-muted-foreground"
  >
    Unauthorized access detected
  </div>
  <div
    class="text-xs text-muted-foreground"
  >
    <div
      class="flex items-center gap-1"
    >
      <div
        data-testid="users-icon"
      />
      <span>
        Assigned to: 
        John Doe
      </span>
      <span
        class="ml-2"
      >
        • 
        2
         units responding
      </span>
    </div>
  </div>
  <div
    class="text-xs text-muted-foreground"
  >
    <div
      class="flex items-center gap-1"
    >
      <div
        data-testid="camera-icon"
      />
      <span>
        Camera 1
      </span>
      <span
        class="ml-2"
      >
        +
        2
         more
      </span>
    </div>
  </div>
  <div
    class="flex items-center gap-1 flex-wrap"
  >
    <div
      data-testid="tag-icon"
    />
    <div
      class="text-xs"
      data-testid="badge"
    >
      security
    </div>
    <div
      class="text-xs"
      data-testid="badge"
    >
      urgent
    </div>
  </div>
</div>
`;
